frontend/app/routes/protected/person-case/parent-details.tsx

```diff
@@ -0,0 +1,389 @@
+
+      const parentAmount = Number(formData.get('parent-amount')) || 0;
+      const inputLength = parentAmount > maxParents ? maxParents : parentAmount;
+
+      const input = Array.from({ length: inputLength }).map((_, i) => {
+        return {
+          unavailable: Boolean(formData.get(`${i}-unavailable`)),
+          givenName: String(formData.get(`${i}-given-name`)),
+          lastName: String(formData.get(`${i}-last-name`)),
+          birthLocation: {
+            country: String(formData.get(`${i}-country`)),
+            province: String(formData.get(`${i}-province`)),
+            city: String(formData.get(`${i}-city`)),
+          },
+        };
+      }) satisfies Partial<ParentDetailsSessionData>;
+
+      const parseResult = v.safeParse(schema, input, { lang });
+
+      if (!parseResult.success) {
+        return data({ errors: v.flatten<typeof schema>(parseResult.issues).nested }, { status: 400 });
+      }
+
+      context.session.inPersonSINCase ??= {};
+      context.session.inPersonSINCase.parentDetails = parseResult.output;
+
+      throw i18nRedirect('routes/protected/person-case/previous-sin.tsx', request);
+    }
+
+    default: {
+      throw new AppError(`Unrecognized action: ${action}`);
+    }
+  }
+}
+
+export default function CreateRequest({ loaderData, actionData, params }: Route.ComponentProps) {
+  const { t } = useTranslation(handle.i18nNamespace);
+  const fetcherKey = useId();
+  const fetcher = useFetcher<Info['actionData']>({ key: fetcherKey });
+  const isSubmitting = fetcher.state !== 'idle';
+  const errors = fetcher.data?.errors;
+  const defaultFormValues = loaderData.defaultFormValues;
+  const [parentAmount, setParentAmount] = useState<number>(defaultFormValues?.length ?? 1);
+
+  return (
+    <>
+      <PageTitle subTitle={t('protected:in-person.title')}>{t('protected:parent-details.page-title')}</PageTitle>
+      <FetcherErrorSummary fetcherKey={fetcherKey}>
+        <fetcher.Form method="post" noValidate>
+          <input type="hidden" name="parent-amount" value={parentAmount} />
+          <ParentInformation
+            amount={parentAmount}
+            setAmount={setParentAmount}
+            defaultFormValues={defaultFormValues}
+            errors={errors}
+          />
+          <div className="mt-8 flex flex-row-reverse flex-wrap items-center justify-end gap-3">
+            <Button name="action" value="next" variant="primary" id="continue-button" disabled={isSubmitting}>
+              {t('protected:person-case.next')}
+            </Button>
+            <Button name="action" value="back" id="back-button" disabled={isSubmitting}>
+              {t('protected:person-case.previous')}
+            </Button>
+          </div>
+        </fetcher.Form>
+      </FetcherErrorSummary>
+    </>
+  );
+}
+
+type detailsType = {
+  unavailable?: boolean;
+  givenName?: string;
+  lastName?: string;
+  country?: string;
+  province?: string;
+  city?: string;
+};
```

Comment:
```suggestion
type FormData = {
  unavailable?: boolean;
  givenName?: string;
  lastName?: string;
  country?: string;
  province?: string;
  city?: string;
};
```

Types are typically defined with their first character uppercased. And I don't think `details` is a good description for this type. As far as I can tell, it represents the values in the form, so why not call it `FormData` or something that fits more with what the data represents?